packages/react-reconciler/src/ReactMutableSource.js

```diff
@@ -30,20 +30,42 @@ export function clearPendingUpdates(
   root: FiberRoot,
   expirationTime: ExpirationTime,
 ): void {
-  if (root.mutableSourcePendingUpdateTime <= expirationTime) {
-    root.mutableSourcePendingUpdateTime = NoWork;
+  if (root.mutableSourceFirstPendingUpdateTime <= expirationTime) {
+    root.mutableSourceFirstPendingUpdateTime = NoWork;
```

Comment:
Needs to account for when the `expirationTime` falls in between the first and last part of the range. Here's the equivalent code for normal updates:

https://github.com/facebook/react/blob/5e464546af4a32c5ae5e704f68bcf4b98a4c440a/packages/react-reconciler/src/ReactFiberRoot.js#L253-L258

Reply:
This change makes sense after we talked about it, although I can't wait for your expiration times refactor now because they make less sense than I thought before 😅